### ISSUE TITLE
Should `abs(::Rational)` check for integer overflow?

### ISSUE THREAD

**Issue by sostock** [JuliaLang/julia#48744]
Arithmetic operations on `Rational`s usually check for overflow, but `abs` doesn’t:
```julia
julia> x = typemin(Int) // 1
-9223372036854775808//1

julia> 2*x # detects overflow and errors
ERROR: OverflowError: 2 * -9223372036854775808 overflowed for type Int64
[...]

julia> abs(x) # doesn’t check for overflow, returns wrong result
-9223372036854775808//1
```

**Comment by vmpyr** [comment]
The documentation says that:

> When abs is applied to signed integers, overflow may occur, resulting in the return of a negative value. This overflow occurs only when abs is applied to the minimum representable value of a signed integer. That is, when x == typemin(typeof(x)), abs(x) == x < 0, not -x as might be expected.

I also think that we can instead detect overflow and return an error, unless there is some reason for going with status quo.

**Comment by mikmoore** [comment]
The quoted documentation on `abs` only discusses signed integers, which `Rational` aren't. Since we check virtually all `Rational` arithmetic, I don't see further harm in checking `abs` as well.

```julia
# from rational.jl
abs(x::Rational) = Rational(abs(x.num), x.den)
```
Replacing the inner `abs` with `checked_abs` looks like it would be sufficient for this.

We should also check other places (at least within `rational.jl`) for dubious use of `abs`. On v1.8.0, the only other one I see is in `hash(x::Rational{<:BitInteger64}, h::UInt)`, where we check `abs(num) < 9007199254740992`. I think this would be fixed by checking `uabs(num) < 9007199254740992`, although I would advocate replacing `9007199254740992` by the equivalent but meaningful `UInt64(maxintfloat(Float64))`.

**Comment by vmpyr** [comment]
makes sense, I will do that asap unless someone beats me to it
